addons/web_editor/static/src/js/snippets.editor.js

```diff
@@ -243,6 +243,11 @@ data.Class = Widget.extend({
                     if (thumbnail) {
                         $div.find('.oe_snippet_thumbnail_img').css('background-image', 'url(' + thumbnail + ')');
                     }
+                    var module_id = $snippet.data("module_id");
+                    if(module_id) {
+                        var install_btn = _.str.sprintf('<div class="o_snippet_btn"><a class="btn btn-primary btn-sm o_install_btn" target="_blank" href="/web#id=%s&view_type=form&model=ir.module.module&action=base.open_module_tree">%s</a> </div>', module_id, _t('Install'));
+                        $div.prepend(install_btn);
+                    }
```

Comment:
Conventions (see https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines) : 
- _camelCase_ : `module_id` -> `moduleID`, ...
- Space after keyword: `if (`
- In your python code above, replace `data-module_id` by `data-module-id`, this follows standard conventions as it allows to do in JS: `.data('moduleId')`

Also, is the `<div class="o_snippet_btn">` useful ?